File: assignments/ruby/wordy/wordy_test.rb
require 'minitest/autorun'
require_relative 'word_problem'

class WordProblemTest < MiniTest::Unit::TestCase
  def test_add_1
    assert_equal 2, WordProblem.new('What is 1 plus 1?').answer
  end

  def test_add_2
    skip
    assert_equal 55, WordProblem.new('What is 53 plus 2?').answer
  end

  def test_add_negative_numbers
    skip
    assert_equal(-11, WordProblem.new('What is -1 plus -10?').answer)
  end

  def test_add_more_digits
    skip
    assert_equal 45801, WordProblem.new('What is 123 plus 45678?').answer
  end

  def test_subtract
    skip
    assert_equal 16, WordProblem.new('What is 4 minus -12?').answer
  end

  def test_multiply
    skip
    assert_equal(-75, WordProblem.new('What is -3 multiplied by 25?').answer)
  end

  def test_divide
    skip
    assert_equal(-11, WordProblem.new('What is 33 divided by -3?').answer)
  end

  def test_add_twice
    skip
    question = 'What is 1 plus 1 plus 1?'
    assert_equal 3, WordProblem.new(question).answer
  end

  def test_add_then_subtract
    skip
    question = 'What is 1 plus 5 minus -2?'
    assert_equal 8, WordProblem.new(question).answer
  end

  def test_subtract_twice
    skip
    question = 'What is 20 minus 4 minus 13?'
    assert_equal 3, WordProblem.new(question).answer
  end

  def test_subtract_then_add
    skip
    question = 'What is 17 minus 6 plus 3?'
    assert_equal 14, WordProblem.new(question).answer
  end

  def test_multiply_twice
    skip
    question = 'What is 2 multiplied by -2 multiplied by 3?'
    assert_equal(-12, WordProblem.new(question).answer)
  end

  def test_add_then_multiply
    skip
    question = 'What is -3 plus 7 multiplied by -2?'
    assert_equal(-8, WordProblem.new(question).answer)
  end

  def test_divide_twice
    skip
    question = 'What is -12 divided by 2 divided by -3?'
    assert_equal 2, WordProblem.new(question).answer
  end

  def test_too_advanced
    skip
    assert_raises ArgumentError do
      WordProblem.new('What is 53 cubed?').answer
    end
  end

  def test_irrelevant
    skip
    assert_raises ArgumentError do
      WordProblem.new('Who is the president of the United States?').answer
    end
  end

end

File: assignments/php/wordy/wordy_test.php
<?php

require realpath(__DIR__ . '/wordy.php');

class WordProblemTest extends PHPUnit_Framework_TestCase {

  /** @test */
  public function test_add_1() {
    $problem = new WordProblem('What is 1 plus 1?');
    $this->assertEquals(2, $problem->answer());
  }

  public function test_add_2() {
    $this->markTestSkipped();
    $problem = new WordProblem('What is 53 plus 2?');
    $this->assertEquals(55, $problem->answer());
  }

  public function test_add_negative_numbers() {
    $this->markTestSkipped();
    $problem = new WordProblem('What is -1 plus -10?');
    $this->assertEquals(-11, $problem->answer());
  }

  public function test_add_more_digits() {
    $this->markTestSkipped();
    $problem = new WordProblem('What is 123 plus 45678?');
    $this->assertEquals(45801, $problem->answer());
  }

  public function test_subtract() {
    $this->markTestSkipped();
    $problem = new WordProblem('What is 4 minus -12?');
    $this->assertEquals(16, $problem->answer());
  }

  public function test_multiply() {
    $this->markTestSkipped();
    $problem = new WordProblem('What is -3 multiplied by 25?');
    $this->assertEquals(-75, $problem->answer());
  }

  public function test_divide() {
    $this->markTestSkipped();
    $problem = new WordProblem('What is 33 divided by -3?');
    $this->assertEquals(-11, $problem->answer());
  }

  public function test_add_twice() {
    $this->markTestSkipped();
    $problem = new WordProblem('What is 1 plus 1 plus 1?');
    $this->assertEquals(3, $problem->answer());
  }

  public function test_add_then_subtract() {
    $this->markTestSkipped();
    $problem = new WordProblem('What is 1 plus 5 minus -2?');
    $this->assertEquals(8, $problem->answer());
  }

  public function test_subtract_twice() {
    $this->markTestSkipped();
    $problem = new WordProblem('What is 20 minus 4 minus 13?');
    $this->assertEquals(3, $problem->answer());
  }

  public function test_subtract_then_add() {
    $this->markTestSkipped();
    $problem = new WordProblem('What is 17 minus 6 plus 3?');
    $this->assertEquals(14, $problem->answer());
  }

  public function test_multiply_twice() {
    $this->markTestSkipped();
    $problem = new WordProblem('What is 2 multiplied by -2 multiplied by 3?');
    $this->assertEquals(-12, $problem->answer());
  }

  public function test_add_then_multiply() {
    $this->markTestSkipped();
    $problem = new WordProblem('What is -3 plus 7 multiplied by -2?');
    $this->assertEquals(-8, $problem->answer());
  }

  public function test_divide_twice() {
    $this->markTestSkipped();
    $problem = new WordProblem('What is -12 divided by 2 divided by -3?');
    $this->assertEquals(2, $problem->answer());
  }

  public function test_too_advanced() {
    $this->markTestSkipped();
    $this->setExpectedException('ArgumentError');

    $problem = new WordProblem('What is 53 cubed?');
    $problem->answer();
  }

  public function test_irrelevant() {
    $this->markTestSkipped();
    $this->setExpectedException('ArgumentError');

    $problem = new WordProblem('Who is the president of the United States?');
    $problem->answer();
  }

}
